docs/reference/README.md

```diff
@@ -1,18 +1,104 @@
-For example, to build a set of Knative API docs for v0.3, you can the tag
-numbers in each repo. So for
-[Serving v0.3.0](https://github.com/knative/serving/tree/v0.3.0) you specify
-`KNATIVE_SERVING_COMMIT="v0.3.0"`, and so on for each of the Build, Eventing,
-and Eventing Sources repos.
+
+## Updating API Reference docs (for Knative maintainers)
+
+The Knative API reference documentation is manually generated using the
+[`gen-api-reference-docs.sh`](../../hack/) tool. If you need to generate a new
+version of the API docs for a recent update or for a new release, you can use
+the steps below.
+
+To learn more about the tool, see the
+[gen-crd-api-reference-docs](https://github.com/ahmetb/gen-crd-api-reference-docs)
+reference page.
+
+### Before you begin
+
+You must meet the following requirements to run the `gen-api-reference-docs.sh`
+tool:
+
+* You need the following software installed:
+  * [`git`](https://git-scm.com/download/)
+  * [`go` version 1.11+](https://golang.org/dl/)
+* You must locally clone the [knative/docs](https://github.com/knative/docs)
+  repository. For example: `git clone git@github.com:knative/docs.git`
+
+### Generating the API
+
+To generate a version of the API:
+
+1. Ensure that your `GOPATH` excludes a value. The `gen-api-reference-docs.sh`
+   script will result in the `GOPATH should not be set` error if your `GOPATH`
+   is configured. You view the value by running the following command:
+
+    ```
+    echo $GOPATH
+    ```
+
+   If your `GOPATH` is already configured, temporarily clear the `GOPATH` value
+   by running the following command:
+
+     ```
+     export GOPATH=""
+     ```
+
+1. Locate the commit or tag numbers that correspond to the version of the API
+   that you want to generate:
+
+   * [Build](https://github.com/knative/build/releases/)
+   * [Eventing](https://github.com/knative/eventing/releases/)
+   * [Eventing Sources](https://github.com/knative/eventing-sources/releases/)
+   * [Serving](https://github.com/knative/serving/releases/)
+
+1. To run the `gen-api-reference-docs.sh` command from the `hack` directory,
+   you specify the commit numbers or tags for each of the corresponding Knative
+   component variables (`KNATIVE_[component_name]_COMMIT`):
+
+    ```
+    KNATIVE_BUILD_COMMIT=[commit_or_tag] \
+    KNATIVE_EVENTING_COMMIT=[commit_or_tag] \
+    KNATIVE_EVENTING_SOURCES_COMMIT=[commit_or_tag] \
+    KNATIVE_SERVING_COMMIT=[commit_or_tag] \
+    ./gen-api-reference-docs.sh
+    ```
+
+    where `[commit_or_tag]` is the commit number or repo tag.
+
+    **Result**
+
+    The `gen-api-reference-docs.sh` tool generates the API in a `tmp` folder.
+    After a successful build, the tool automatically opens that folder
+    in the `tmp` directory.
+
+1. Copy the generated API files into the `docs/reference` directory of your
+   knative/docs clone.
+
+You can now perform the necessary steps to open a PR, complete a review, and
```

Comment:
Should this be a step too?

Reply:
i feel like these steps should be known if you are a "maintainer" but ive added a link to the info in the contributor docs

Reply:
Edits are here: https://github.com/knative/docs/pull/1010